app_dart/test/request_handlers/check_for_waiting_pull_requests_test.dart

```diff
@@ -0,0 +1,402 @@
+      log = FakeLogging();
+      githubGraphQLClient = FakeGraphQLClient();
+      flutterRepoPRs.clear();
+      engineRepoPRs.clear();
+      PullRequestHelper._counter = 0;
+
+      githubGraphQLClient._mutateResultForOptions =
+          (MutationOptions options) => QueryResult();
+
+      githubGraphQLClient._queryResultForOptions = (QueryOptions options) {
+        expect(options.variables['sOwner'], 'flutter');
+        expect(options.variables['sLabelName'],
+            config.waitingForTreeToGoGreenLabelNameValue);
+
+        final String repoName = options.variables['sName'];
+        if (repoName == 'flutter') {
+          return createQueryResult(flutterRepoPRs);
+        } else if (repoName == 'engine') {
+          return createQueryResult(engineRepoPRs);
+        } else {
+          fail('unexpected repo $repoName');
+        }
+      };
+
+      tester = ApiRequestHandlerTester(request: request);
+      config.waitingForTreeToGoGreenLabelNameValue =
+          'waiting for tree to go green';
+      config.githubGraphQLClient = githubGraphQLClient;
+
+      handler = CheckForWaitingPullRequests(
+        config,
+        auth,
+        loggingProvider: () => log,
+      );
+    });
+
+    void _verifyQueries() {
+      githubGraphQLClient.verifyQueries(
+        <QueryOptions>[
+          QueryOptions(
+            document: labeledPullRequestsWithReviewsQuery,
+            fetchPolicy: FetchPolicy.noCache,
+            variables: <String, dynamic>{
+              'sOwner': 'flutter',
+              'sName': 'flutter',
+              'sLabelName': config.waitingForTreeToGoGreenLabelNameValue,
+            },
+          ),
+          QueryOptions(
+            document: labeledPullRequestsWithReviewsQuery,
+            fetchPolicy: FetchPolicy.noCache,
+            variables: <String, dynamic>{
+              'sOwner': 'flutter',
+              'sName': 'engine',
+              'sLabelName': config.waitingForTreeToGoGreenLabelNameValue,
+            },
+          ),
+        ],
+      );
+    }
+
+    test('Merges first PR in list, all successful', () async {
+      flutterRepoPRs.add(PullRequestHelper());
+      flutterRepoPRs.add(PullRequestHelper()); // will be ignored.
+      engineRepoPRs.add(PullRequestHelper());
+
+      await tester.get(handler);
+
+      _verifyQueries();
+
+      githubGraphQLClient.verifyMutations(
+        <MutationOptions>[
+          MutationOptions(
+            document: mergePullRequestMutation,
+            variables: <String, dynamic>{
+              'id': flutterRepoPRs.first.id,
+              'oid': 'deadbeef',
+            },
+          ),
+          MutationOptions(
+            document: mergePullRequestMutation,
+            variables: <String, dynamic>{
+              'id': engineRepoPRs.first.id,
+              'oid': 'deadbeef',
+            },
+          ),
+        ],
+      );
+    });
+
+    test('Ignores PRs that are too new', () async {
```

Comment:
Can we please also have a test around `now - 50 minutes` and `now - 70 minutes` to verify our threshold of `now - 60 minutes`?

Reply:
Done.